src/components/GameView/GameOverDialog.vue

```diff
@@ -90,6 +124,27 @@ export default {
 
       return 'loss-img';
     },
+    currentMatch(){
+      return this.game.currentMatch
+    },
+    matchGameStats(){
+      if(!this.game.currentMatch?.games) return []
+      const result= this.game.currentMatch.games.map(game=>{
+        if (game.result === -1) return 'I'
+        if(game.result === 0 || game.result === 1){
+          if(game.p0 === this.player.id) return 'W'
+          return 'L'
+        }
+        return 'D'
```

Comment:
This nest of ifs is really confusing, it would be preferable to block them out a bit more legibly. Combing returns if you can would look good. Ideally, some sort of map or case statement (and @itsalaidbacklife knows I hate those) might even be better.
```suggestion
if (game.result < 0) {
  return 'I';
}

if ([0, 1].includes(game.result)) {
  return game.p0 === this.player.id ?
    'W' :
    'L';
}

return 'D';
```

Reply:
@seriouslysean I think the includes block is wrong because the `game.result` value is the pNum of whichever player won the game so we need to differentiate between whether the player is p0 or p1 differently based on whether it's 0 or 1. I think this is a good use case for switch: one separate return statement per possible value of a specified variable

```suggestion
        switch(game.result) {
          case 0: // p0 won game
            return game.p0 === this.player.id ? 'W' : 'L';
          case 1: // p1 won game
            return game.p1 === this.player.id ? 'W' : 'L';;
          case 2: // draw
            return 'D';
          default: // incomplete
            return 'I';
        }
```

Reply:
I wasn’t entirely sure and tired to match it as best I could. This is one of those cases a switch actually makes sense so I’m in to it.